src/pv.rs

```diff
@@ -61,6 +62,62 @@ impl PvPacket {
             private: std::ptr::null_mut(),
         }
     }
+
+    // check whether the new header can be attached to payload from the start of packet
+    fn is_addable(header_length: usize, packet: &PvPacket) -> bool {
+        let total_len: u32 = packet.size - packet.start;
```

Comment:
Why? `packet.start` cannot be decreased? There **are** headroom for **prepending header**

Reply:
The changed code will check if a header can be attached or not